src/Tizen.Network.Stc/Tizen.Network.Stc/StcRule.cs

```diff
@@ -0,0 +1,373 @@
+                if (ret != (int)StcError.None)
+                {
+                    Log.Error(Globals.LogTag, "Failed to set Interface type, Error - " + (StcError)ret);
+                    StcErrorFactory.ThrowStcException(ret);
+                }
+            }
+        }
+
+        /// <summary>
+        /// A property for Time period for statistics rule.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        /// <value>Time period.</value>
+        /// <privilege>http://tizen.org/privilege/network.get</privilege>
+        /// <exception cref="NotSupportedException">Thrown while setting this property when Stc is not supported.</exception>
+        /// <exception cref="InvalidOperationException">Thrown while setting this value due to an invalid operation.</exception>
+        public TimePeriodType TimePeriod
+        {
+            get
+            {
+                TimePeriodType timePeriod;
+                int ret = Interop.Stc.Rule.GetTimePeriod(_ruleHandle, out timePeriod);
+                if (ret != (int)StcError.None)
+                {
+                    Log.Error(Globals.LogTag, "Failed to get Time period, Error - " + (StcError)ret);
+                }
+                return timePeriod;
+            }
+            set
+            {
+                if (_disposed)
+                {
+                    throw new ObjectDisposedException("Invalid StcRule instance (Object may have been disposed or released)");
+                }
+                int ret = Interop.Stc.Rule.SetTimePeriod(_ruleHandle, value);
+                if (ret != (int)StcError.None)
+                {
+                    Log.Error(Globals.LogTag, "Failed to set Time period, Error - " + (StcError)ret);
+                    StcErrorFactory.ThrowStcException(ret);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets the statistics information an application matched rule asynchronously.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        /// <param name="infoCb">The callback is called for each application that used network in between timestamps specified.</param>
+        /// <privilege>http://tizen.org/privilege/network.get</privilege>
+        /// <exception cref="NotSupportedException">Thrown when the Stc is not supported.</exception>
+        /// <exception cref="UnauthorizedAccessException">Thrown when the permission is denied.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when the method failed due to an invalid operation.</exception>
+        public void GetStats(StatisticsCallback infoCb)
+        {
+            if (_disposed)
+            {
+                throw new ObjectDisposedException("Invalid StcRule instance (Object may have been disposed or released)");
+            }
+
+            _getStatsObjectCb = infoCb;
+
+            _getStatsCb = (int result, IntPtr info, IntPtr userData) =>
```

Comment:
info handle only valid in callback,
Handle is invalid on outside of callback

So, you need to pack all data into c# layer

Reply:
Now clone() CAPI is available, and i am using it this time.